### ISSUE TITLE
Query Parameters all have to be Strings or Lists of Strings

### ISSUE THREAD

**Issue by keaganhilliard** [audiobookshelf-flutter/audiobookshelf_api#57]
Fun fact about http calls in dart. They will fail if pass anything other than a string or list of strings into the query parameters.

```dart
class BoolBinaryConverter implements JsonConverter<bool, int> {
  const BoolBinaryConverter();

  @override
  bool fromJson(int json) => json == 1;

  @override
  int toJson(bool object) => object ? 1 : 0;
}
```

This causes `items.get` requests to fail. To use this in a query parameter it has to either be '1' or '0'. So you could potentially explicitly call toString() on any parameters or coerce parameter attributes to strings.

**Comment by k9withabone** [comment]
Huh, guess I didn't read the `Uri` docs close enough. I don't `BoolBinaryConverter` is the place to do this though. It is also used in places where an `int` is needed. I'll fix this in `AudiobookshelfApi.request()`.

**Comment by k9withabone** [comment]
Fixed in #58, @keaganhilliard could you take a look at it when you get a chance.
